blocks/api/test/registration.js

```diff
@@ -82,7 +82,7 @@ describe( 'blocks', () => {
 		it( 'should accept valid block names', () => {
 			const block = registerBlockType( 'my-plugin/fancy-block-4', defaultBlockSettings );
 			expect( console ).not.toHaveErrored();
-			expect( block ).toEqual( {
+			expect( block ).toMatchObject( {
```

Comment:
My goal in this change was to simply test block registration, but block registration hooks are already registered at this point and affect the block settings. Do you agree, and if so, do you have any suggestions about how to handle this?

One possibility I see but don't love is to remove all `blocks.registerBlockType` filters and expect unit tests for those filters to register and remove them. This may require additional exports from hook modules for the sake of testing.